pkg/controller/deployment_util.go

```diff
@@ -731,6 +747,31 @@ func GetNewMachineSet(ctx context.Context, deployment *v1alpha1.MachineDeploymen
 	return FindNewMachineSet(deployment, rsList), nil
 }
 
+func filterMachinesWithUpdateSuccessfulLabel(machines []*v1alpha1.Machine) []*v1alpha1.Machine {
+	machinesWithUpdateSuccessfulLabel := make([]*v1alpha1.Machine, 0)
+	for _, machine := range machines {
+		if labelValue, ok := machine.Labels[v1alpha1.LabelKeyNodeUpdateResult]; ok && labelValue == v1alpha1.LabelValueNodeUpdateSuccessful {
+			cond := getMachineCondition(machine, v1alpha1.NodeInPlaceUpdate)
+			// only consider machines with the update successful condition
+			if cond != nil && cond.Reason == v1alpha1.UpdateSuccessful {
```

Comment:
Please check only labels as they should represent the definite case that machines are successfully updated